pkg/agent/cluster.go

```diff
@@ -0,0 +1,330 @@
+	RestAPIClusterStatusPreparingForInstallationSeen    bool
+	RestAPIClusterStatusReadySeen                       bool
+	RestAPICurrentClusterStatus                         string
+	RestAPIPreviousClusterStatus                        string
+	RestAPIHostValidationsPassed                        bool
+	ClusterKubeAPISeen                                  bool
+}
+
+// NewCluster initializes a Cluster object
+func NewCluster(ctx context.Context, assetDir string) (*Cluster, error) {
+
+	czero := &Cluster{}
+	capi := &clientSet{}
+
+	restclient, err := NewNodeZeroRestClient(ctx, assetDir)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+	kubeclient, err := NewClusterKubeAPIClient(ctx, assetDir)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+
+	capi.Rest = restclient
+	capi.Kube = kubeclient
+
+	clusterID, err := restclient.getClusterID()
+	if err != nil {
+		return nil, err
+	}
+	clusterInfraEnvID, err := restclient.getClusterInfraEnvID()
+	if err != nil {
+		return nil, err
+	}
+
+	cinstallstatushistory := &clusterInstallStatusHistory{
+		RestAPISeen:                                         false,
+		RestAPIClusterStatusAddingHostsSeen:                 false,
+		RestAPIClusterStatusCancelledSeen:                   false,
+		RestAPIClusterStatusInstallingSeen:                  false,
+		RestAPIClusterStatusInstallingPendingUserActionSeen: false,
+		RestAPIClusterStatusInsufficientSeen:                false,
+		RestAPIClusterStatusFinalizingSeen:                  false,
+		RestAPIClusterStatusErrorSeen:                       false,
+		RestAPIClusterStatusPendingForInputSeen:             false,
+		RestAPIClusterStatusPreparingForInstallationSeen:    false,
+		RestAPIClusterStatusReadySeen:                       false,
+		RestAPICurrentClusterStatus:                         "",
+		RestAPIPreviousClusterStatus:                        "",
+		RestAPIHostValidationsPassed:                        false,
+		ClusterKubeAPISeen:                                  false,
+	}
+
+	czero.Ctx = ctx
+	czero.API = capi
+	czero.clusterID = clusterID
+	czero.clusterInfraEnvID = clusterInfraEnvID
+	czero.installHistory = cinstallstatushistory
+	return czero, nil
+}
+
+// Get Retrieve the current cluster metadata from the Agent Rest API
+func (czero *Cluster) Get() (*models.Cluster, error) {
+	// GET /v2/clusters/{cluster_zero_id}
+	getClusterParams := &installer.V2GetClusterParams{ClusterID: *czero.clusterID}
+	result, err := czero.API.Rest.Client.Installer.V2GetCluster(czero.Ctx, getClusterParams)
+	if err != nil {
+		return nil, err
+	}
+	cluster := result.Payload
+	return cluster, nil
+}
+
+// IsBootstrapComplete Determine if the cluster that agent installer
+// is installing has completed the bootstrap process.
+func (czero *Cluster) IsBootstrapComplete() (bool, error) {
+
+	agentRestAPILive, agentRestAPIErr := czero.API.Rest.IsRestAPILive()
+	if agentRestAPIErr != nil {
+		logrus.Debug("node zero Agent API is not available")
```

Comment:
Just a suggestion to use wrap here and at line 124 to save a line, e.g.
logrus.Debug(errors.Wrap(agentRestAPIErr, "node zero Agent API is not available"))